js/fitbit-service.js

```diff
@@ -0,0 +1,185 @@
+                method: 'GET',
+            }, function(err, body, token) {
+                if (err) return next(err);
+                var profile = JSON.parse(body);
+                // if token is not null, a refesh has happened and we need to persist the new token
+                if (token)
+                    persist.write(tfile, token, function(err) {
+                        if (err) return next(err);
+                            res.send('<pre>' + JSON.stringify(profile, null, 2) + '</pre>');
+                    });
+                else
+                    res.send('<pre>' + JSON.stringify(profile, null, 2) + '</pre>');
+            });
+        });
+
+        var port = process.env.PORT || 4000;
+        console.log('express is listening on: ', port);
+        app.listen(port);
+
+        // Read the persisted token, initially captured by a webapp.
+        //
+        fs.stat(tfile, function(err, stat) {
+            if(err == null) {
+                console.log('Fitbit token File exists');
+
+                persist.read(tfile, function(err, token) {
+                    if (err) {
+                        console.log('persist read error: ', err);
+                    }
+
+                    // Set the client's token
+                    fitbit.setToken(token);
+                });
+            } else if(err.code == 'ENOENT') {
+                console.log('Error reading Fitbit token file! This might be the first time you are running the app, if so, make sure you browse to http://yourappurl:yourport/fitbit - this will redirect you to the auth page.', err);
+            } else {
+                console.log(err);
+            }
+        });
+
+        service.profileSummary = function(callback) {
+            if(service.summary === null){
+                return null;
+            }
+
+            // Make an API call to get the fitbit user's profile data
+            fitbit.request({
+                uri: "https://api.fitbit.com/1/user/-/profile.json",
+                method: 'GET',
+            }, function( err, body, token ) {
+                if (err) {
+                    console.log(err);
+
+                }
+                var result = JSON.parse(body);
+
+                // If the token arg is not null, then a refresh has occured and
+                // we must persist the new token.
+                if (token) {
+                    persist.write( tfile, token, function(err) {
+                        if (err) console.log(err);
+                    });
+                }
+                return callback(result.user);
+            });
+        }
+
+        service.todaySummary = function(callback) {
+            if(service.today === null){
+                return null;
+            }
+
+            var today = new Date();
+            var dd = today.getDate();
+            var mm = today.getMonth()+1; //January is 0!
+            var yyyy = today.getFullYear();
```

Comment:
You should use `moment` here (see the calendar service for examples) - it makes rendering these things on the mirror way easier (and gives you localization for free).